src/components/QrScanner/QrScanner.tsx

```diff
@@ -0,0 +1,316 @@
+  };
+  const [qrCode, setQrCode] = useState(defaultQrCode);
+  const [qrCodeText, setQrCodeText] = useState<string>("");
+  const [qrScanStage, setQrScanStage] = useState(QR_SCAN_STAGE.SCANNING);
+  const [cameraFacingMode, setCameraFacingMode] = useState(
+    CAMERA_FACING_MODE.BACK
+  );
+  const [checkInName, setCheckInName] = useState("none");
+  const [error, setError] = useState("");
+
+  useEffect(() => {
+    if (
+      !qrCode ||
+      qrCodeText === "" ||
+      typeof qrCodeText !== "string" ||
+      qrScanStage !== QR_SCAN_STAGE.SCANNING
+    )
+      return;
+
+    // data is arranged: email;event_id;year
+    // id is the array of the data split by ";"
+    const id = qrCodeText.split(";");
+    const userID = id[0];
+    const eventIDAndYear = id[1] + ";" + id[2];
+    const userFName = id[3];
+
+    if (!isCheckInValid(id, userID, eventIDAndYear)) {
+      return;
+    }
+
+    checkInUser(userID, userFName);
+  }, [qrCodeText]);
+
+  const scanStateClassName = (scanStage: string) => {
+    switch (scanStage) {
+      case QR_SCAN_STAGE.SUCCESS:
+        return "bg-secondary-color";
+
+      case QR_SCAN_STAGE.SCANNING:
+        return "bg-secondary-color";
+
+      case QR_SCAN_STAGE.FAILED:
+        return "bg-[#cc0000]";
+
+      default:
+        return "bg-secondary-color";
+    }
+  };
+
+  const isCheckInValid = (
+    id: string[],
+    userID: string,
+    eventIDAndYear: string
+  ): boolean => {
+    if (eventIDAndYear !== event.id + ";" + event.year) {
+      cycleQrScanStage(QR_SCAN_STAGE.FAILED, 8000);
+
+      // if there are not 4 and first item is not an email, then the QR code is invalid
+      if (id.length !== 4 && !emailCheck(userID)) {
+        setError("Invalid BizTech QR code.");
+      } else {
+        setError("Please check that your QR code is for this event.");
+      }
+      return false;
+    }
+
+    const user = rows?.filter((row) => row.id === userID)[0];
+
+    if (!user) {
+      cycleQrScanStage(QR_SCAN_STAGE.FAILED, 6000);
+      setError("Person is not registered for this event.");
+      return false;
+    }
+
+    // get the person's name
+    setCheckInName(
+      `${user.firstName ? user.firstName : user.fname} ${
+        user.lastName ? user.lastName : user.lname
+      } (${userID})`
+    );
+
+    // If the user is already checked in, show an error
+    if (user.registrationStatus === REGISTRATION_STATUS.CHECKED_IN) {
+      cycleQrScanStage(QR_SCAN_STAGE.FAILED, 5000);
+      setError("Person is already checked in.");
+      return false;
+    } else if (user.registrationStatus === REGISTRATION_STATUS.CANCELLED) {
+      cycleQrScanStage(QR_SCAN_STAGE.FAILED, 5000);
+      setError("Person had their registration cancelled. Cannot check-in.");
+      return false;
+    } else if (user.registrationStatus === REGISTRATION_STATUS.WAITLISTED) {
+      cycleQrScanStage(QR_SCAN_STAGE.FAILED, 5000);
+      setError("Person is on the waitlist. Cannot check-in.");
+      return false;
```

Comment:
nit: we could refactor all uses of `cycleQrScanStage` where we pass in a 'magic number' (the number of milliseconds to delay) and replace them with a Symbolic Constant export https://refactoring.guru/replace-magic-number-with-symbolic-constant